hop/auth.py

```diff
@@ -336,6 +337,15 @@ def read_new_credential(csv_file=None):
         if len(password) == 0:
             raise RuntimeError("Password may not be empty")
         hostname = input("Hostname (may be empty): ")
+        # check that the user entered a sensible hostname, possibly with a port
+        name_re = re.compile("^(kafka://)?([^:/]*(:[0-9]*)?)$")
+        match = name_re.match(hostname)
+        if match is None:
+            raise RuntimeError("Unable to parse hostname. "
+                               "Please enter either `hostname` or `hostname:port`.")
+        if match.group(1) is not None:
+            logger.info(f"Ignoring '{match.group(1)}' prefix on hostname")
+        hostname = match.group(2)
```

Comment:
It may be better for this to be extracted into a separate function where the validation can be tested in isolation. It seems like `read_new_credential` is doing a bit too much now. The input sanitation may also be of use in other places as well, for example if a credential is not added via `hop auth add` but instead one of the existing credentials are modified.

Reply:
Done. This isn't currently applied to data read from a CSV file, on the grounds that we're supposed to generate those with valid data, aand there's not any way to modify a credential, but if we add one this can be used for it.

Reply:
That sounds reasonable to me.